packages/wdio-browserstack-service/src/Percy/PercyBinary.ts

```diff
@@ -0,0 +1,214 @@
+import url from 'node:url'
+import yauzl from 'yauzl'
+
+import { createRequire } from 'node:module'
+const require = createRequire(import.meta.url)
+
+const fs = require('node:fs')
+const { https } = require('follow-redirects')
+
+import path from 'node:path'
+import os from 'node:os'
+import { spawn } from 'node:child_process'
+import { PercyLogger } from './PercyLogger.js'
+
+class PercyBinary {
+    #hostOS = process.platform
+    #httpPath: any = null
+    #binaryName = 'percy'
+
+    #orderedPaths = [
+        path.join(this.#homedir(), '.browserstack'),
+        process.cwd(),
+        os.tmpdir()
+    ]
+
+    constructor() {
+        const base = 'https://github.com/percy/cli/releases/latest/download'
+        if (this.#hostOS.match(/darwin|mac os/i)) {
+            this.#httpPath = base + '/percy-osx.zip'
+        } else if (this.#hostOS.match(/mswin|msys|mingw|cygwin|bccwin|wince|emc|win32/i)) {
+            this.#httpPath = base + '/percy-win.zip'
+            this.#binaryName = 'percy.exe'
+        } else {
+            this.#httpPath = base + '/percy-linux.zip'
+        }
+    }
+
+    #homedir(): any {
+        if (typeof os.homedir === 'function') {
+            return os.homedir()
+        }
+
+        const env = process.env
+        const home = env.HOME
+        const user = env.LOGNAME || env.USER || env.LNAME || env.USERNAME
+
+        if (process.platform === 'win32') {
+            return env.USERPROFILE || (env.HOMEDRIVE || 'null') + env.HOMEPATH || home || null
+        }
+
+        if (process.platform === 'darwin') {
+            return home || (user ? '/Users/' + user : null)
+        }
+
+        if (process.platform === 'linux') {
+            return home || (process.getuid && process.getuid() === 0 ? '/root' : (user ? '/home/' + user : null))
+        }
+
+        return home || null
+    }
+
+    #makePath(path: string) {
+        try {
+            if (!this.#checkPath(path)) {
+                fs.mkdirSync(path)
+            }
+            return true
+        } catch {
+            return false
+        }
+    }
```

Comment:
import the fs promise package (`import fsp from 'node:fs/promises'`) and do

```suggestion
    #makePath(path: string) {
        if (this.#checkPath(path)) {
            return true
        }
        return fs.mkdir(path).then(() => true, () => false)
    }
```

Reply:
Done